src/Events/Admin/Onboarding/Controller.php

```diff
@@ -88,6 +88,59 @@ public function handle_onboarding_page_dismiss(): void {
 		$this->container->make( Landing_Page::class )->handle_onboarding_page_dismiss();
 	}
 
+	/**
+	 * Redirects users to the Guided Setup page when accessing any TEC settings or management page for the first time.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function redirect_tec_pages_to_guided_setup(): void {
+		// Do not redirect if they are already on the Guided Setup page.
+		$page = tec_get_request_var( 'page' );
+		if ( Landing_Page::$slug === $page ) {
+			return;
+		}
+
+		// Do not redirect if they have been to the Guided Setup page already.
+		if ( (bool) tribe_get_option( 'tec_onboarding_wizard_visited_guided_setup', false ) ) {
+			return;
+		}
+
+		// Do not redirect if they dismissed the Guided Setup page.
+		if ( (bool) Landing_Page::is_dismissed() ) {
```

Comment:
This is type hinted in the function that it returns bool as a result it CANNOT return anything else. So we don't need to cast it.

check the method [here](https://github.com/the-events-calendar/the-events-calendar/blob/b47be5bd162dd2ac06f44d928b4fcf9d0fbfd40e/src/Events/Admin/Onboarding/Landing_Page.php#L110)